### PR TITLE
Adding in additional metrics to track for database connections

### PR DESCRIPTION
**Pull request by InfernoJJ**
Adding in additional metrics to track for database connections

### CODE REVIEW

File: database/mysql.go [hunk 1]
```diff
@@ -92,6 +104,11 @@ func (my *mysql) Connect(ctx context.Context) (*sql.DB, error) {
 				my.connections.With("state", "idle").Set(float64(stats.Idle))
 				my.connections.With("state", "inuse").Set(float64(stats.InUse))
 				my.connections.With("state", "open").Set(float64(stats.OpenConnections))
+				my.counters.With("stat", "wait_count").Set(float64(stats.WaitCount))
```
Comment (adamdecaf) [inline database/mysql.go:107]: This is called `counter` in the metric definition.

Comment (InfernoJJ) [reply database/mysql.go]: Yup, just caught that myself too.

File: database/mysql.go [hunk 2]
```diff
@@ -38,6 +38,17 @@ var (
 		Help: "How many MySQL connections and what status they're in.",
 	}, []string{"state"})
 
+	mysqlConnectionsCounters = kitprom.NewGaugeFrom(stdprom.GaugeOpts{
+		Name: "mysql_connections_counters",
```
Comment (adamdecaf) [inline database/mysql.go:42]: Shouldn't this be `_gauges` ?

Comment (InfernoJJ) [reply database/mysql.go:42]: I was going off the naming much like how the status one is going off of it. 
These are counter under the hood but the API for prometheus go doesn't allow me to _set_ a value on a counter.

Comment (adamdecaf) [reply database/mysql.go:42]: Right, gauges record the value at lots of points in time. Counters only increment. https://prometheus.io/docs/concepts/metric_types/#gauge

Prometheus namings try and refer some type-level information about the data. `_seconds` or `_count`

Comment (InfernoJJ) [reply database/mysql.go:42]: yeah hence the counter metric is labelled `wait_ms`

Comment (InfernoJJ) [reply database/mysql.go:42]:
```			wait_count: The total number of connections waited for.
			wait_duration: The total time blocked waiting for a new connection.
			max_idle_closed: The total number of connections closed due to SetMaxIdleConns.
			max_idle_time_closed: The total number of connections closed due to SetConnMaxIdleTime.
			max_lifetime_closed: The total number of connections closed due to SetConnMaxLifetime.```

**Review by adamdecaf — APPROVED** [review]

**Pull request merged**